### ISSUE TITLE
scnetworkgateway.exe cannot find sccoreerr.dll

### ISSUE THREAD

**Issue by hentil** [Starcounter/Starcounter#2671]
![image](https://cloud.githubusercontent.com/assets/11957872/7154645/7e0d86e4-e358-11e4-9b02-af1a1f8439f1.png)

Installed this version
http://downloads.starcounter.com/download/Starcounter/NextNightlyBuilds/latest

try to run using shortcut.
File doesn't exist on disk

**Comment by alemoi** [comment]
Thanks for report! The problem seems very strange like the installation is corrupted. Which shortcut do you use to start the program? Are there any specific way to reproduce the situation?

**Comment by hentil** [comment]
Starcounter Personal Server

**Comment by alemoi** [comment]
The reason is probably a not updated white list file for Next Starcounter.

**Comment by k-rus** [comment]
It might be good idea to run the acceptance test on next.

**Comment by dan31** [comment]
@k-rus, yes, we will definitely do!

**Comment by parherman** [comment]
I finished a first test run earlier today with the acceptance test on next, and it looks like it fails because of this issue

**Comment by per-samuelsson** [comment]
Assigning this to me to fix, since we need progress on it for Heads.

**Comment by hentil** [comment]
With version 239 I can open Starcounter personal server och create a database. However, when starting my applikation I get the following error message:

C:\Source\Research\SC2\heads.next\src\Heads.Application\bin\Debug>star -d=heads Heads.exe
Heads -> heads (starting database .)

[14:52:06, Starcounter (heads)]
Access violation (C0000005) detected at C:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.239.8\Level0\src\sccoredb\export\sc
coredb.cpp:3649 (Thread 5908).

[14:52:04, Starcounter (heads)]
System.DllNotFoundException: Det gick inte att läsa in DLL-filen systables.dll: Det går inte att hitta den angivna modul
en. (Undantag från HRESULT: 0x8007007E)
   vid Starcounter.Internal.systables.star_prepare_system_tables()
   vid Starcounter.Hosting.Package.UpdateDatabaseSchemaAndRegisterTypes(TypeDef[] unregisteredTypeDefs) i c:\TeamCity\Te
amCity8\buildAgent\work\sc-2.0.239.8\Level1\src\Hosting\Starcounter.Hosting\Package.cs:rad 283
   vid Starcounter.Hosting.Package.ProcessWithinCurrentApplication(Application application) i c:\TeamCity\TeamCity8\buil
dAgent\work\sc-2.0.239.8\Level1\src\Hosting\Starcounter.Hosting\Package.cs:rad 219
   vid Starcounter.Hosting.Package.Process() i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.239.8\Level1\src\Hosting\Sta
rcounter.Hosting\Package.cs:rad 144
   vid StarcounterInternal.Hosting.Processor.RunMessageLoop(Void* hsched) i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0
.239.8\Level1\src\Hosting\Starcounter.Hosting\Processor.cs:rad 70
TypeName=
HResult=-2146233052

**Comment by per-samuelsson** [comment]
Thanks for reporting again, @hentil. And sorry.

We have decided to make a little extension to our build system, adding a "blacklist" log file to artifacts, to make it easier to see what files are not included. I will do this, take a look at the result, and then try to identifiy if its even more files that are excluded that should not be, and eventually fix it.

I'll keep you updated!

**Comment by per-samuelsson** [comment]
I'm running a custom build with a fix for this. It will be available on the relevant download channel shortly (versioned 2.0.241.8). Let us know how it turns out!

**Comment by hentil** [comment]
Following error when trying to start with version 2.0.241.8.

C:\Source\Research\SC2\heads.next\src\Heads.Application\bin\Debug>star -d=heads Heads.exe
Heads -> heads (starting database ...)

[10:53:05, Starcounter (heads)]
Access violation (C0000005) detected at C:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level0\src\sccoredb\export\sc
coredb.cpp:3649 (Thread 1112).

**Comment by per-samuelsson** [comment]
> Following error when trying to start with version 2.0.241.8.

That one is more tricky I'm afraid. Access violations are harder to track.

Before continuing, I want to stress that this is version **next**, i.e. a version that not considered stable at all, and which is very much work in progress at the moment. :smile: 

Back to topic. Can you take a look if there is some hints in the error log? Run the following to copy the last 5 warnings (and errors) to the clipboard, and paste the result here. Thanks.

> staradmin list --max=5 log warnings | clip

**Comment by hentil** [comment]
[10:53:05, Critical, Starcounter (heads)]
Access violation (C0000005) detected at C:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level0\src\sccoredb\export\sccoredb.cpp:3649 (Thread 1112).

[10:53:06, Error, Starcounter.Server.Commands (scadminserver)]
Starcounter.DbException: ScErrDatabaseEngineTerminated (SCERR10007): The server detected that the user code host executable (sccode.exe) unexpectedly terminated during a management operation. sccode.exe, PID=7124, Database=heads, Exitcode=1069
Version: 2.0.241.8.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR10007.
   vid Starcounter.Server.DatabaseEngine.WaitUntilCodeHostOnline(Process codeHostProcess, Database database) i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level1\src\Server\Starcounter.Server\CoreServices\DatabaseEngine.cs:rad 370
   vid Starcounter.Server.Commands.Processors.StartDatabaseCommandProcessor.<>c__DisplayClass4.<Execute>b__3(CommandTask task) i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level1\src\Server\Starcounter.Server\Commands\Processors\StartDatabaseCommandProcessor.cs:rad 82
   vid Starcounter.Server.Commands.CommandProcessor.<>c__DisplayClass1.<WithinTask>b__0(CommandTask t) i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level1\src\Server\Starcounter.Server\Commands\CommandProcessor.cs:rad 484
   vid Starcounter.Server.Commands.CommandProcessor.WithinTask(CommandTask task, Func`2 func) i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level1\src\Server\Starcounter.Server\Commands\CommandProcessor.cs:rad 514
   vid Starcounter.Server.Commands.CommandProcessor.ProcessCommand(NotifyCommandStatusChangedCallback notifyStatusChangedCallback) i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level1\src\Server\Starcounter.Server\Commands\CommandProcessor.cs:rad 300
ErrorCode=10007
HResult=-2146233088
HelpLink=https://github.com/Starcounter/Starcounter/wiki/SCERR10007
---> System.Exception: ScErrUnhandledException (SCERR1069): An unhandled exception was detected.
Version: 2.0.241.8.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR1069.
HResult=-2146233088
HelpLink=https://github.com/Starcounter/Starcounter/wiki/SCERR1069


[10:53:06, Error, Starcounter.Server.Commands (scadminserver)]
ScErrServerCommandFailed (SCERR10011): The operation "Starting database heads." failed.
Version: 2.0.241.8.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR10011.

[10:54:06, Error, Starcounter.Server.Commands (scadminserver)]
Starcounter.DbException: ScErrDatabaseNotFound (SCERR10002): A database with the specified name was not found. Database: 'sc://headsutv32.innowait.se/personal/default'.
Version: 2.0.241.8.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR10002.
   vid Starcounter.Server.Commands.Processors.StartDatabaseCommandProcessor.Execute() i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level1\src\Server\Starcounter.Server\Commands\Processors\StartDatabaseCommandProcessor.cs:rad 40
   vid Starcounter.Server.Commands.CommandProcessor.ProcessCommand(NotifyCommandStatusChangedCallback notifyStatusChangedCallback) i c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.241.8\Level1\src\Server\Starcounter.Server\Commands\CommandProcessor.cs:rad 300
ErrorCode=10002
HResult=-2146233088
HelpLink=https://github.com/Starcounter/Starcounter/wiki/SCERR10002


[10:54:06, Error, Starcounter.Server.Commands (scadminserver)]
ScErrServerCommandFailed (SCERR10011): The operation "Starting database default." failed.
Version: 2.0.241.8.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR10011.

**Comment by hentil** [comment]
Ok, so this might be connected to the migration project?

**Comment by per-samuelsson** [comment]
> Ok, so this might be connected to the migration project?

I'm currently setting up a local environment, to see if I can reproduce. When I run it on a local build (using latest sources), I don't get the problem. Now I'm swtiching to go the exact same path as you, i.e. downloading the exact version and installing it. I'll let you know as soon as I get it settled.

**Comment by per-samuelsson** [comment]
> I'm currently setting up a local environment, to see if I can reproduce.

I could not. On my machine, doing the exact same thing as you did works fine. The application start without a problem.

Can you do a test for me, @hentil? Do this

> staradmin kill all
> move "%starcounterbin%\editionlibraries" "%starcounterbin%\editionlibraries_deleted"
> star -d=foo Heads.exe

And where "foo" is a placeholder for any name really, just make sure its a database name you haven't used before.

Now, do you get the same error?

**Comment by hentil** [comment]
That solved the problem! My app is up and running.

**Comment by per-samuelsson** [comment]
> That solved the problem! My app is up and running.

OK, sweet.

**Comment by per-samuelsson** [comment]
> That solved the problem! My app is up and running.

So now we have got to a point where things seem to work decent, minus the fact that you get crashes because there are polyjuice edition stuff that interfere with any standard Starcounter application. Which is bad, but its appearently a decision that was made, and it has to be worked around manually every time you are reinstalling Starcounter (renamed folder as I described above).

With this, I'm closing this issue. The polyjuice stuff will be adressed elsewhere, and by someone else.
